### PR TITLE
Adding diff script

### PR DESCRIPTION
**Pull request by Noctunus**
Added script to help with diffs against other branches with the version structure we've decided on

### CODE REVIEW

File: scripts/diff_against_dev.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## * Check if a file with version-1 exists in the origin branch
+## * Do a diff against the other file
+
+CUR_BRANCH=$(git branch --show-current)
+ORIGIN=${1:-dev}
+
+git fetch origin $ORIGIN
+
+function check_file {
+	FILE=$1
```
Comment (havan) [inline scripts/diff_against_dev.sh]: `$1` is used for both the `ORIGIN` and `FILE` variables?

**Review by havan — COMMENTED** [review]
Can you explain how the script will be used?

File: scripts/diff_against_dev.sh (hunk 1 above)
Comment (Noctunus) [reply scripts/diff_against_dev.sh]: $1 in the function refers to the parameter passed to the function not the script parameter

**Comment by Noctunus** [comment]
> Can you explain how the script will be used?

Execute the script in the base directory of a branch and it will do the diff against dev and show the difference of the added files to the existing files in dev with version-1.

This is described in the header of the script - or is that not clear by the comment?

File: scripts/diff_against_dev.sh (hunk 1 above)
Comment (havan) [reply scripts/diff_against_dev.sh]: 🤦🏼‍♂️ my dislike to bash scripting shows

**Comment by havan** [comment]
> This is described in the header of the script - or is that not clear by the comment?

I ran it and got no result but I then realized that it only checks added files.

**Review by havan — APPROVED** [review]

**Pull request merged**